sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -452,6 +452,31 @@ def test_max_feature_regression():
     assert_true(deviance < 0.5, "GB failed with deviance %.4f" % deviance)
 
 
+def test_feature_importance_regression():
+    # Test to make sure Gini importance is calculated correctly
+    # Make sure that results are summed, then normalized to avoid
+    # over-weighting features from later stages
+    boston = datasets.load_boston()
+    X, y = shuffle(boston.data, boston.target, random_state=13)
```

Comment:
I don't think that you need to shuffle.
You could shuffle directly inside the train_test_split